kernel/base/build.gradle

```diff
@@ -39,7 +39,9 @@ dependencies {
   compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.6'
   compile group: 'org.apache.commons', name: 'commons-text', version: '1.1'
   compile group: 'org.apache.maven.shared', name: 'maven-invoker', version: '3.0.0'
-  compile group: 'org.slf4j', name: 'slf4j-simple', version: '1.7.22'
+  compile group: 'org.slf4j', name: 'slf4j-api', version: '1.7.25'
+  compile group: 'ch.qos.logback', name: 'logback-classic', version: '1.2.3'
```

Comment:
we need to stick with slf4j, not add logback.

Reply:
Why do you say that? slf4j-api provides the logging API, so the way that users log a message doesn't change at all as a result of this PR. It's only the way that you configure logging that changes (with this change `logback.xml` is used to configure the logger).

From what I've seen, probably the most recommended way to do logging in Java is to use slf4j-api + logback. I contribute quite a bit to Play Framework, which is a popular Java web framework, and it probably took us a few releases over a couple of years to get logging to a place where it really worked well, We ended up using slf4j-api + logback there, so this seems like a fairly good change to me, since it aligns well with what I've seen from a good amount of experience with Java logging

Reply:
i believe the log should go to stderr, so it appears with the rest of the jupyter log.  SimpleLogger in slf4j already supports that so another lib is not needed.

Reply:
These seem like separate issues. We have just as many config files in the current version of the PR with slf4j-simple as we did in the logback version of the PR. I saw your comment that we should try to reduce the number of config files. I'm guessing if we can do that for slf4j-simple then we can for logback as well. Also, logback can send logs to stderr. In terms of libraries, it's the same number of libraries on the classpath in both cases. I would generally prefer using logback to slf4j-simple still